### PR TITLE
feat(codex-ui): Tabbar component

### PR DESCRIPTION
**Pull request by e11sy**
## Tabbar component implemented
image

emits in tabbar also tested with console.log()
image

### CODE REVIEW

File: codex-ui/src/vue/components/tabbar/index.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export type * from './Tab.types';
 import Tab from './Tab.vue';
+import Tabbar from './Tabbar.vue';
```
Comment (slaveeks) [inline codex-ui/src/vue/components/tabbar/index.ts:3]:
```suggestion
import TabBar from './TabBar.vue';
```

File: codex-ui/src/vue/components/tabbar/Tabbar.vue [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+<script setup lang="ts">
+import { TabList, TabParams } from './Tab.types';
+import Tab from './Tab.vue';
+
+defineEmits<{
+  /* eslint-disable-next-line no-unused-vars */
+  (e: 'tabDiscard', identifier: TabParams['identifier']): void;
+  /* eslint-disable-next-line no-unused-vars */
+  (e: 'tabClicked', identifier: TabParams['identifier']): void;
+}>();
```
Comment (neSpecc) [inline codex-ui/src/vue/components/tabbar/Tabbar.vue:24]: lets use the new vue 3.3. [syntax](https://vuejs.org/guide/typescript/composition-api.html#typing-component-emits)

```
const emit = defineEmits<{
  click: [tab: TabParams]
  discard: [tab: TabParams]
}>()
```

File: codex-ui/src/vue/components/tabbar/Tabbar.vue [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+</script>
+
+<style land="postcss">
+.tabbar {
+  --min-width: calc(var(--v-padding) * 2 + var(--size-icon));
+  display: flex;
+  grid-template-columns: repeat(auto-fit, minmax(var(--min-width), auto));
+  padding: 0px var(--v-padding);
+  width: 100%;
+  height: 44px;
```
Comment (neSpecc) [inline codex-ui/src/vue/components/tabbar/Tabbar.vue:39]: hardcoded

File: codex-ui/src/vue/components/tabbar/Tab.vue [hunk 4]
```diff
@@ -26,9 +27,11 @@
           />
         </div>
       </template>
-      {{ title }}
+      <div :class="[$style['tab__body-text']]">
+        {{ title }}
+      </div>
       <Icon
-        v-if="closable"
+        v-if="closable && isNarrow"
```
Comment (elizachi) [inline codex-ui/src/vue/components/tabbar/Tab.vue:34]: Maybe rename isNarrow to more clear?

Comment (e11sy) [reply codex-ui/src/vue/components/tabbar/Tab.vue:34]: any suggestions? (this variable checks that cross icon could be placed in available space)

Comment (elizachi) [reply codex-ui/src/vue/components/tabbar/Tab.vue:34]: maybe `isFit`?

File: codex-ui/src/vue/components/tabbar/index.ts (hunk 1 above)
Comment (e11sy) [reply codex-ui/src/vue/components/tabbar/index.ts:3]: i think tastes differs

File: codex-ui/src/vue/components/tabbar/Tab.vue (hunk 4 above)
Comment (e11sy) [reply codex-ui/src/vue/components/tabbar/Tab.vue]: isNarrow now deprecated

**Review by elizachi — APPROVED** [review]

**Review by neSpecc — APPROVED** [review]

**Review by slaveeks — APPROVED** [review]

**Pull request merged**